internal/controller/drplacementcontrol_controller.go

```diff
@@ -1237,36 +1230,26 @@ func (r *DRPlacementControlReconciler) finalizeDRPC(ctx context.Context, drpc *r
 		return fmt.Errorf("failed to get DRPolicy while finalizing DRPC (%w)", err)
 	}
 
-	drClusters, err := GetDRClusters(ctx, r.Client, drPolicy)
-	if err != nil {
-		return fmt.Errorf("failed to get drclusters. Error (%w)", err)
-	}
-
-	// Verify VRGs have been deleted
-	vrgs, _, _, err := getVRGsFromManagedClusters(r.MCVGetter, drpc, drClusters, vrgNamespace, log)
-	if err != nil {
-		return fmt.Errorf("failed to retrieve VRGs. We'll retry later. Error (%w)", err)
+	// cleanup for Volsync, VRG and MCVs
+	if err = r.cleanupForVolSyncVRGMCV(ctx, drPolicy, log, mwu, drpc, vrgNamespace); err != nil {
+		return fmt.Errorf("%w", err)
```

Comment:
Just return err here, no need to format it again. Same comment applies to other such patterns in the change.